runelite-mixins/src/main/java/net/runelite/mixins/RSFriendMixin.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2018, UniquePassive <https://github.com/uniquepassive>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.mixins;
+
+import com.google.common.cache.Cache;
+import net.runelite.api.mixins.FieldHook;
+import net.runelite.api.mixins.Inject;
+import net.runelite.api.mixins.Mixin;
+import net.runelite.api.mixins.Shadow;
+import net.runelite.rs.api.RSFriend;
+
+@Mixin(RSFriend.class)
+public abstract class RSFriendMixin implements RSFriend
+{
+	@Shadow("cachedFriends")
+	private static Cache<String, Boolean> cachedFriends;
+
+	@Shadow("cachedOnlineFriends")
+	private static Cache<String, Boolean> cachedOnlineFriends;
+
+	@FieldHook("previousName")
+	@Inject
+	public void previousNameChanged(int idx)
+	{
+		if (cachedFriends != null)
+		{
+			cachedFriends.invalidate(getPreviousName());
```

Comment:
Why is changing the previous name invalidating the friendcache, when it is only looked up by current name?

Reply:
If a player changes name, it invalidates the previous name so that if somebody else takes their previous name, they won't be cached. Although I think that maybe never happens in the real world..